django/db/backends/base/schema.py

```diff
@@ -893,6 +906,24 @@ def _alter_column_type_sql(self, model, old_field, new_field, new_type):
             [],
         )
 
+    def _alter_column_collation_sql(self, new_field, new_type, new_collation):
+        if new_field.db_collation is None:
+            return (
+                self.sql_alter_column_no_collate % {
+                    'column': self.quote_name(new_field.column),
+                    'type': new_type,
+                },
+                [],
+            )
+        return (
+            self.sql_alter_column_collate % {
+                'column': self.quote_name(new_field.column),
+                'type': new_type,
+                'collation': self.quote_name(new_collation),
```

Comment:
I wonder why we quote a collation name in all backends when altering a field, but we do this only on PostgreSQL in table creation :thinking:

Reply:
I think it should be quoted in all cases.

Reply:
I was mostly going by what was in the vendor docs. Quoting works everywhere except SQLite though, so I'll special case it there.